lib/measured/conversion_builder.rb

```diff
@@ -0,0 +1,35 @@
+class Measured::ConversionBuilder
+  def initialize
+    @base_unit = nil
+    @units = []
+    @case_sensitive = false
+  end
+
+  def base(unit_name, aliases: [])
+    raise Measured::UnitError, "Setting #{unit_name} as base unit, but already defined as #{@base_unit}." if @base_unit
+    @base_unit = Measured::Unit.new(unit_name, aliases: aliases)
+  end
+
+  def unit(unit_name, aliases: [], value:)
+    @units << Measured::Unit.new(unit_name, aliases: aliases, value: value)
+  end
+
+  def case_sensitive(is_case_sensitive)
+    @case_sensitive = is_case_sensitive
+  end
+
+  def conversion
+    raise Measured::UnitError, "A base unit has not yet been set." unless @base_unit
+    check_for_duplicate_unit_names
```

Comment:
When is `conversion` first called? Inside the `build` method? I'm ok with it being called after everything is attempted to be added on two conditions:
* That it doesn't allow some overwrite state by accident.
* That it is called on env initialization, so that apps know they are in an inconsistent state on load, not first use.